pkg/controller/certificatesigningrequests/controller.go

```diff
@@ -124,6 +148,17 @@ func (c *Controller) Register(ctx *controllerpkg.Context) (workqueue.RateLimitin
 	csrInformer.Informer().AddEventHandler(&controllerpkg.QueuingEventHandler{Queue: c.queue})
 	issuerInformer.Informer().AddEventHandler(&controllerpkg.BlockingEventHandler{WorkFunc: c.handleGenericIssuer})
 
+	// Ensure we catch extra informers that are owned by certificate signing requests
+	for _, i := range c.extraInformers {
+		i.AddEventHandler(&controllerpkg.BlockingEventHandler{
+			WorkFunc: controllerpkg.HandleOwnedResourceNamespacedFunc(c.log, c.queue,
+				schema.GroupVersionKind{Version: "v1", Group: "certificates.k8s.io", Kind: "CertificateSigningRequest"},
+				func(_, name string) (interface{}, error) {
+					return c.csrLister.Get(name)
+				}),
+		})
+	}
```

Comment:
suggestion: there is a lot going on here, would it be possible to have a test for that? E.g., when a parent ACME Issuer is updated, it triggers a resync of the CertificateSigningRequest it owns.

Reply:
This is tested by e2e, read comment below.

Reply:
Please share the test code you mentioned in the call :smile:

Reply:
https://github.com/jetstack/cert-manager/blob/5ebcbced5ca92efd1377b6f8520b4837901d76e6/pkg/controller/certificate-shim/gateways/controller_test.go#L38